types/got/got-tests.ts

```diff
@@ -266,20 +266,24 @@ got.extend({ method: 'POST' })('/example');
 got.extend({ method: 'POST' }).extend({ headers: {} }).stream('/example');
 
 // JSON options:
-// $ExpectType Promise<any>
+got.extend({ timeout: 100 })('/example', { body: {}, json: true });
+got.extend({ timeout: 100 })('/example', { body: { foo: 'bar' }, json: true });
+// $ExpectType Promise<string>
 got.extend({ json: true })('/example').then(({ body }) => body);
-// $ExpectType Promise<any>
+// $ExpectType Promise<string>
```

Comment:
This is incorrect. If you pass `json: true` option, then result must `Promise<any>`, because `JSON.parse` will be called.